### PR TITLE
Authentication option doesn't take effect if there's no authorization policy 

### LINKED ISSUE #3380: Authentication option doesn't take effect if there's no authorization policy
When running opa as a server, if you specify the `--authentication` option without the `--authorization=basic` option, the authentication never takes effect.

While this is spelled out in the [docs](https://www.openpolicyagent.org/docs/latest/security/#authentication-and-authorization), it would be beneficial to also have opa emit a warning message at runtime to make it clear that no authentication is happening due to the lack of a authz policy.

### PR DESCRIPTION
**Pull request by kale-amruta**
This fix gives a Error msg when you start the opa runtime server with authentication as TOKEN and authorization OFF

Fixes #3380

### CODE REVIEW

File: runtime/runtime.go
Comment (anderseknert) [inline runtime/runtime.go:413]: This should be reverted.

File: runtime/runtime_test.go
Comment (anderseknert) [inline runtime/runtime_test.go:307]: Is this really a test helper?

File: runtime/runtime.go
Comment (anderseknert) [inline runtime/runtime.go:309]: The server doesn't really have a "warn" log level (only debug, info, and error) so this would either be info or error. Given how this is most likely a rare event I would probably go with error level.

File: runtime/runtime_test.go
Comment (kale-amruta) [reply runtime/runtime_test.go]: my bad. Removed it

File: runtime/runtime_test.go
Comment (anderseknert) [inline runtime/runtime_test.go:308]: A bit of a nit, but OPA logs to stderr, so this should either be renamed `stderr` or I guess simply just `output` as we're really just printing to a buffer for the test.

**Review by anderseknert — COMMENTED** [review]
Starting to look good! :)

**Review by anderseknert — DISMISSED** [review]
LGTM. You'll need to squash the commits before merge though. Thanks @kale-amruta!

**Review by anderseknert — APPROVED** [review]

**Comment by kale-amruta** [comment]
@anderseknert Thanks! I dont think I have access to merge the PR. Will you be merging it?

**Comment by anderseknert** [comment]
It looks good from my POV, but never hurts to get a second opinion before merge so leaving this for someone else to review.

**Comment by kale-amruta** [comment]
@anderseknert sure, makes sense.

File: runtime/runtime.go
Comment (ashutosh-narkar) [inline runtime/runtime.go:309]: Nit: Should we log a warning (ie. `logrus.Warn`) ?

Comment (anderseknert) [reply runtime/runtime.go:309]: Better check the resolved conversations ;)

**Pull request merged**
